packages/labs/analyzer/src/lib/javascript/packages.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {AnalyzerInterface, PackageInfo, PackageJson} from '../model.js';
+import {AbsolutePath} from '../paths.js';
+
+/**
+ * Starting from a given module path, searches up until the nearest package.json
+ * is found, returning that folder. If none is found, an error is thrown.
+ */
+const getPackageRootForModulePath = (
+  modulePath: AbsolutePath,
+  analyzer: AnalyzerInterface
+): AbsolutePath => {
+  // TODO(kschaaf): Add caching & invalidation
+  const {fs, path} = analyzer;
+  let searchDir = modulePath as string;
+  const root = path.parse(searchDir).root;
+  while (searchDir !== root) {
+    if (fs.fileExists(path.join(searchDir, 'package.json'))) {
+      return searchDir as AbsolutePath;
+    }
+    searchDir = path.dirname(searchDir);
+  }
+  throw new Error(`No package.json found for module path ${modulePath}`);
+};
+
+/**
+ * Reads and parses a package.json file contained in the given folder.
+ */
+const getPackageJsonFromPackageRoot = (
+  packageRoot: AbsolutePath,
+  analyzer: AnalyzerInterface
+): PackageJson => {
+  // TODO(kschaaf): Add caching & invalidation
+  const {fs, path} = analyzer;
+  const packageJson = fs.readFile(path.join(packageRoot, 'package.json'));
+  if (packageJson !== undefined) {
+    return JSON.parse(packageJson) as PackageJson;
+  }
+  throw new Error(`No package.json found at ${packageRoot}`);
+};
+
+/**
+ * Returns an analyzer `PackageInfo` model for the nearest package of the given
+ * path.
+ */
+export const getPackageInfo = (
```

Comment:
Pretty much all the helpers that take `analyzer` as an argument could just be instance methods on the Analyzer class, but then we'd end up with a giant Analyzer class rather than a bunch of coherently collated helpers. If they're for internal use (not for the Analyzer user's use), I'm inclined to keep the current factoring.